src/fdk_fulltext_search/ingest/__init__.py

```diff
@@ -492,3 +508,77 @@ def init_info_doc(index_name: str, now: str) -> None:
     if not es_client.indices.exists("info"):
         es_client.indices.create(index="info")
     es_client.index(index="info", body=init_doc)
+
+
+def create_or_update_dataset_pipeline(pipeline_id: str) -> None:
+    try:
+        es_client.ingest.get_pipeline(pipeline_id)
+        es_client.ingest.delete_pipeline(pipeline_id)
+    except NotFoundError:
+        pass
+
+    es_client.ingest.put_pipeline(
+        pipeline_id,
+        {
+            "processors": [
+                {
+                    "script": {
+                        "description": "Concatinate fdkFormat type and code",
+                        "lang": "painless",
+                        "source": """
+                        ArrayList formats = new ArrayList();
+                        if(ctx['distribution'] != null){
+                            for (Map distr : ctx['distribution']) {
+                                if(distr['fdkFormat'] != null){
+                                    for (Map fmt : distr['fdkFormat']) {
+                                        if(fmt['fdkType'] == 'UNKNOWN') {
+                                            formats.add('UNKNOWN')
+                                        } else {
+                                            formats.add(fmt['fdkType'] + ' ' + fmt['code'])
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        ctx["fdkFormatPrefixed"] = formats
+                      """,
+                    }
+                }
+            ]
+        },
+    )
+
+
+def create_or_update_dataservice_pipeline(pipeline_id: str) -> None:
+    try:
+        es_client.ingest.get_pipeline(pipeline_id)
+        es_client.ingest.delete_pipeline(pipeline_id)
+    except NotFoundError:
+        pass
+
+    es_client.ingest.put_pipeline(
+        pipeline_id,
+        {
+            "processors": [
+                {
+                    "script": {
+                        "description": "Concatinate fdkFormat type and code",
+                        "lang": "painless",
+                        "source": """
+                        ArrayList formats = new ArrayList();
```

Comment:
Dette er duplisert kode som brukes likt både for dataset og dataservice, bør lages som en funksjon som da vedlikeholdes et sted.